l5kit/l5kit/simulation/dataset.py

```diff
@@ -46,19 +46,34 @@ def __init__(self, scene_dataset_batch: Dict[int, EgoDataset], sim_cfg: Simulati
         if not len(scene_dataset_batch):
             raise ValueError("can't build a simulation dataset with an empty batch")
         self.scene_dataset_batch: Dict[int, EgoDataset] = scene_dataset_batch
+        self.sim_cfg = sim_cfg
 
-        self._len = min([len(scene_dt.dataset.frames) for scene_dt in self.scene_dataset_batch.values()])
+        # we must limit the scenes to the part which will be simulated
+        # we cut each scene so that it starts from there and ends after `num_simulation_steps`
+        start_frame_idx = self.sim_cfg.start_frame_index
+        if self.sim_cfg.num_simulation_steps is None:
+            end_frame_idx = min([len(scene_dt.dataset.frames) for scene_dt in self.scene_dataset_batch.values()])
+        else:
```

Comment:
Do we want to support when the number of simulation steps is greater than the minimum length ? I think it would be better to add a condition and raise an exception for now.

Reply:
it will raise a `ValueError` later when trying to cut the zarr, but I can add a check here sure